libsolidity/formal/CHC.cpp

```diff
@@ -152,7 +152,7 @@ void CHC::endVisit(ContractDefinition const& _contract)
 	smtutil::Expression zeroes(true);
 	for (auto var: stateVariablesIncludingInheritedAndPrivate(_contract))
 		zeroes = zeroes && currentValue(*var) == smt::zeroValue(var->type());
-	addRule(smtutil::Expression::implies(initialConstraints(_contract) && zeroes, predicate(entry)), entry.functor().name);
+	addRule(smtutil::Expression::implies((initialConstraints(_contract) && zeroes) && state().txConstraints(nullptr), predicate(entry)), entry.functor().name);
```

Comment:
```suggestion
	addRule(smtutil::Expression::implies((initialConstraints(_contract) && zeroes) && state().txConstraints(), predicate(entry)), entry.functor().name);
```
after the other suggestion

Reply:
Fixed.